relayer/src/substrate/beefy_subscription.rs

```diff
@@ -0,0 +1,195 @@
+use crate::relay::beefy_syncer::BeefySyncer;
+use crate::{prelude::*, relay::justification::BeefyJustification};
+use beefy_primitives::VersionedFinalityProof;
+use futures::Stream;
+use futures::StreamExt;
+use sp_runtime::traits::{Header as HeaderT, UniqueSaturatedInto};
+
+use super::BlockNumber;
+
+const BEEFY_MIN_DELTA: u32 = 8;
+
+pub async fn get_commitment_from_block<T>(
+    sub: &SubUnsignedClient<T>,
+    block: BlockNumber<T>,
+    is_mandatory: bool,
+) -> AnyResult<Option<BeefyJustification<T>>>
+where
+    T: SenderConfig,
+{
+    trace!("Get commitment from block {:?}", block);
+    let hash = sub.block_hash(Some(block)).await?;
+    let block = sub
```

Comment:
It could be useful to have methods `sub.block_by_num(<number>)` and `sub.block_by_hash(<hash>)`